pkg/testutil/testutil.go

```diff
@@ -17,15 +17,33 @@ package testutil
 import (
 	"net/url"
 	"runtime"
+	"strings"
 	"testing"
+	"time"
 )
 
-// WARNING: This is a hack.
-// Remove this when we are able to block/check the status of the go-routines.
-func ForceGosched() {
-	// possibility enough to sched up to 10 go routines.
-	for i := 0; i < 10000; i++ {
-		runtime.Gosched()
+func runnableGoroutineNumber() int {
```

Comment:
I agree but I just don't know of any other way to get the level of introspection. My alternative/initial suggestion to Yicheng was to inject another dummy channel into all the potential blocking points that is used only in tests.